### ISSUE TITLE
Style/IfWithBooleanLiteralBranches: multiple branches readability

### ISSUE THREAD

**Issue by BuonOmo** [rubocop/rubocop#11006]
## Is your feature request related to a problem? Please describe.

The below method would in my opinion loose in readability if it were corrected by rubocop:

```ruby
      def contains?(rhs, opts = {})
        if Feature::Geometry === rhs
          contains?(BoundingBox.new(@factory).add(rhs))
        elsif rhs.empty?
          true
        elsif empty?
          false
        elsif @min_x > rhs.min_x || @max_x < rhs.max_x || @min_y > rhs.min_y || @max_y < rhs.max_y
          false
        elsif @has_m && rhs.has_m && !opts[:ignore_m] && (@min_m > rhs.min_m || @max_m < rhs.max_m)
          false
        elsif @has_z && rhs.has_z && !opts[:ignore_z] && (@min_z > rhs.min_z || @max_z < rhs.max_z)
          false
        else
          true
        end
      end
```
<sup>

[source](https://github.com/rgeo/rgeo/blob/b2fe667df6c71db0637fdfbf647881d5c780a4e4/lib/rgeo/cartesian/bounding_box.rb#L282-L298)

</sup>

The branching helps understanding clearly which result is expected for which conditions. And in [the doc examples](https://docs.rubocop.org/rubocop/cops_style.html#styleifwithbooleanliteralbranches) we can see that it is well suited for simple if-else cases.

## Describe the solution you'd like

Having an option that allows for multiple branches to stay as it is:

```yaml
Style/IfWithBooleanLiteralBranches:
  Enabled: true
  MultipleBranches: false
```

## Describe alternatives you've considered

disabling the rule, but this would make me miss the simple if-else which are to be removed.

**Comment by koic** [comment]
I'm the author of this cop. TBH, I didn't consider this case when I made this. So, I think it's good to allow when multiple `elsif`s are used instead of the new option. OTOH, if there is only single `elsif`, `elsif` will not exist due to autocorrection. So I think the current behavior can be kept. I've opened #11007.
